### PR TITLE
Report: deal with unsupported timezones. Fixes #1056

### LINKED ISSUE #1056: Unsupported time zone specified undefined
Lighthouse Version: 1.2.1
Chrome Version: 54.0.2840.99 
Error Message: Unexpected token , in JSON at position 45
Stack Trace:

> RangeError: Unsupported time zone specified undefined
    at new DateTimeFormat (native)
    at ReportGenerator._formatTime (chrome-extension://blipmdconlkpinefehnmjammfjpmpbjk/scripts/lighthouse-background.js:10838:23)
    at ReportGenerator.generateHTML (chrome-extension://blipmdconlkpinefehnmjammfjpmpbjk/scripts/lighthouse-background.js:10970:27)
    at window.createReportPageAsBlob (chrome-extension://blipmdconlkpinefehnmjammfjpmpbjk/scripts/lighthouse-background.js:38312:28)
    at connection.getCurrentTabURL.then.then.results (chrome-extension://blipmdconlkpinefehnmjammfjpmpbjk/scripts/lighthouse-background.js:38282:30)

### PR DESCRIPTION
**Pull request by ebidel**
R: @brendankenny @patrickhulce @paulirish 

Fixes #1056

### CODE REVIEW

File: lighthouse-core/report/report-generator.js [hunk 1]
```diff
@@ -132,7 +132,14 @@ class ReportGenerator {
       hour: 'numeric', minute: 'numeric', second: 'numeric',
       timeZoneName: 'short'
     };
-    const formatter = new Intl.DateTimeFormat('en-US', options);
+    let formatter = new Intl.DateTimeFormat('en-US', options);
+
+    // Force UTC if runtime timezone could not be detected.
+    // See https://github.com/GoogleChrome/lighthouse/issues/1056
+    if (formatter.resolvedOptions().timeZone === 'Etc/Unknown') {
```
Comment (wardpeet) [inline lighthouse-core/report/report-generator.js]: perhaps it's better to `formatter.resolvedOptions().timeZone.toLowerCase() === 'etc/unknown'`
not sure if chrome will ever change this.

Comment (ebidel) [inline lighthouse-core/report/report-generator.js]: Made it even more liberal :)

File: lighthouse-core/report/report-generator.js [hunk 2]
```diff
@@ -132,7 +132,14 @@ class ReportGenerator {
       hour: 'numeric', minute: 'numeric', second: 'numeric',
       timeZoneName: 'short'
     };
-    const formatter = new Intl.DateTimeFormat('en-US', options);
+    let formatter = new Intl.DateTimeFormat('en-US', options);
+
+    // Force UTC if runtime timezone could not be detected.
+    // See https://github.com/GoogleChrome/lighthouse/issues/1056
+    if (formatter.resolvedOptions().timeZone.match(/Unknown/i)) {
```
Comment (brendankenny) [inline lighthouse-core/report/report-generator.js]: I think you should still check for `"Etc"`. That's an [important part of the zone name](https://en.wikipedia.org/wiki/Tz_database#Area), I just learned, and I'm not entirely sure that you can't have an `Unknown` location inside other Areas.

Comment (ebidel) [inline lighthouse-core/report/report-generator.js]: Done. PTAL

**Pull request merged**